Board.java

```diff
@@ -0,0 +1,184 @@
+		private Game.Direction goHere = Game.defaultDirection;
+		
+		@Override
+		public Location getLocation() {
+			return this.location;
+		}
+
+		@Override
+		public void setLocation(Location location) {
+			this.location = location;
+		}
+
+		@Override
+		public Game.Direction getDirection() {
+			return this.direction;
+		}
+
+		@Override
+		public void setDirection(Game.Direction direction) {
+			this.direction = direction;
+		}
+
+		public Game.Direction getNextDirection() {
+			return goHere;
+		}
+
+		public void setNextDirection(Game.Direction goHere) {
+			this.goHere = goHere;
+		}
+	}
+	
+	private Creature[][] board;
+	private int ghostsNumber;
+	
+	public Creature[][] get() {
+		return board.clone();
+	}
+
+	public Creature get(Location l){
+		return board[l.getX()][l.getY()];
+	}
+	
+	public void set(Location l, Creature c){
+		board[l.getX()][l.getY()] = c;
+		
+		if (c == Creature.Wall || c == Creature.Null || c == Creature.Point)
+			return;
+		
+		c.setLocation(l);
+		
+		if (c.getDirection() == null)
+			c.setDirection(Game.defaultDirection);
+	}
+	
+	public void limitToSpecificCreatures(Creature[] CreaturesForThisBoard){
+		for (int i = 0; i < board.length; i++){
+			for (int j = 0; j < board[0].length; j++){
+				if (!isCreatureAllowed( get(new Location(i, j) ), CreaturesForThisBoard ))
+					set( new Location(i, j), Creature.Null );
+			}
+		}
+	}
+	
+	public int getDimensions(){
+		if (board.length != board[0].length)
+			return Integer.parseInt( new String( board.length + "" + board[0].length ) );
+		
+		return board.length;
+	}
+	
+	public int getGhostNum(){
+		return this.ghostsNumber;
+	}
+	
+	private Creature[][] setBoard() throws FileNotFoundException{
+		Scanner sc = new Scanner(new BufferedReader(new FileReader("./Board.txt")));
+	    while(sc.hasNextLine()) {
+	    	for (int i = 0; i < board.length && sc.hasNextLine(); i++) {
+    			String[] line = sc.nextLine().trim().split("	");
+	        	for (int j = 0; j < line.length; j++)
+					initialize( new Location(i, j), StringToCreature(line[j]) );
+	        }
+	    }
+	    sc.close();
+	    return board;
+	}
+	
+	private Creature StringToCreature(String st){
+		switch(st){
+			case "P":
+				return Creature.Pacman;
+			case "W":
+				return Creature.Wall;
+			case "-":
+				return Creature.Point;
+			case "G":				
+				this.ghostsNumber++;
+				switch(this.ghostsNumber){
+					case 1:
+						return Creature.Ghost1;
+					case 2:
+						return Creature.Ghost2;
+					case 3:
+						return Creature.Ghost3;
+					case 4:
+						return Creature.Ghost4;
+				}
+				
+			default:
+				return Creature.Null;
+		}
+	}
+
+	private Creature[][] setBoardWithStringArray(String[] sts) {
+		
+		for (int i = 0; i < sts.length; i++){
+			for (int j = 0; j < sts[i].length(); j++)
+				initialize( new Location(i, j), StringToCreature(sts[i].charAt(j) + "") );
+		}
+		
+		return this.board;
+	}
+
+	private boolean isCreatureAllowed(Board.Creature isAllowed, Board.Creature[] allowedArray){

```

Comment:
in Java 8 and above, you can use Arrays util to do it in 1 line of code - 
Arrays.stream(allowedArray).anyMatch(isAllowed::equals)